src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -218,11 +224,9 @@ private BuildListener getListener() {
         // Un-synchronized to prevent deadlocks (combination of run and metadataGuard)
         // Note that in portions where multithreaded access is possible we are already synchronizing on metadataGuard
```

Comment:
I'm not so sure about this comment. There are cases like [this](https://github.com/jenkinsci/workflow-job-plugin/blob/94fa4b91dcf3fbefc4310f97d18ba80eb5372c12/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L461-L463) where the comment looks accurate, but there are also cases like [this](https://github.com/jenkinsci/workflow-job-plugin/blob/94fa4b91dcf3fbefc4310f97d18ba80eb5372c12/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L402) which are clearly happening from various threads with no synchronization, and I don't want to break things by adding synchronization to make the comment accurate.

I think that in practice the listener is initialized very early when the `WorkflowRun` is created, and no other threads are acting on the `WorkflowRun` at that point, and then since `listener` is volatile we never really execute the body of the if statement on the next line until after the build completes, so we don't have to worry about double-initialization.